src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -159,40 +171,47 @@ data Enqueue =
 -- of the next layer up to configure these peers as desired.
 --
 -- TODO: Sanity check the number of forwarding sets and number of alternatives.
--- TODO: Verify the max queue sizes against the updated policy.
 type EnqueuePolicy nid =
            MsgType nid  -- ^ Type of the message we want to send
         -> [Enqueue]
 
--- TODO: Verify the policy for data requests
-defaultEnqueuePolicy :: NodeType           -- ^ Type of this node
-                     -> EnqueuePolicy nid
-defaultEnqueuePolicy NodeCore = go
+-- | Pick default policy given node type
+--
+-- NOTE: Assumes standard behind-NAT node in the case of edge nodes.
+defaultEnqueuePolicy :: NodeType -> EnqueuePolicy nid
+defaultEnqueuePolicy NodeCore  = defaultEnqueuePolicyCore
+defaultEnqueuePolicy NodeRelay = defaultEnqueuePolicyRelay
+defaultEnqueuePolicy NodeEdge  = defaultEnqueuePolicyEdgeBehindNat
+
+-- | Default enqueue policy for code nodes
+defaultEnqueuePolicyCore :: EnqueuePolicy nid
+defaultEnqueuePolicyCore = go
   where
-    -- Enqueue policy for core nodes
     go :: EnqueuePolicy nid
     go (MsgAnnounceBlockHeader _) = [
         EnqueueAll NodeCore  (MaxAhead 0) PHighest
-      , EnqueueAll NodeRelay (MaxAhead 0) PMedium
+      , EnqueueAll NodeRelay (MaxAhead 0) PHigh
       ]
     go MsgRequestBlockHeaders = [
-        EnqueueAll NodeCore  (MaxAhead 20) PLowest
-      , EnqueueAll NodeRelay (MaxAhead 20) PLowest
+        EnqueueAll NodeCore  (MaxAhead 1) PHigh
+      , EnqueueAll NodeRelay (MaxAhead 1) PHigh
       ]
-    -- TODO use the argument (Set nid).
     go (MsgRequestBlock _) = [
```

Comment:
Actually this TODO and the other equivalent TODOs are in fact done. The `EnqueueOne [NodeRelay, NodeCore]` bit is exactly what says that we have to send to one of the supplied node ids (though preferring relays over core nodes and not permitting requests to edge nodes except perhaps by fallback behaviour).